src/java.base/share/classes/sun/security/ssl/SSLTransport.java

```diff
@@ -122,16 +123,42 @@ static Plaintext decode(TransportContext context,
         } catch (AEADBadTagException bte) {
             throw context.fatal(Alert.BAD_RECORD_MAC, bte);
         } catch (BadPaddingException bpe) {
-            /*
-             * The basic SSLv3 record protection involves (optional)
-             * encryption for privacy, and an integrity check ensuring
-             * data origin authentication.  We do them both here, and
-             * throw a fatal alert if the integrity check fails.
-             */
-             Alert alert = (context.handshakeContext != null) ?
-                     Alert.HANDSHAKE_FAILURE :
-                     Alert.BAD_RECORD_MAC;
-            throw context.fatal(alert, bpe);
+            // Check for unexpected plaintext alert message during TLSv1.3 handshake.
+            // This can happen if client doesn't receive ServerHello due to network timeout
+            // and tries to close the connection by sending an alert message.
+            if (srcsLength == 1 && !context.sslConfig.isClientMode && !context.isNegotiated &&
+                    context.handshakeContext != null &&
+                    context.handshakeContext.negotiatedProtocol.useTLS13PlusSpec()) {
+                ByteBuffer packet = srcs[srcsOffset].duplicate();
+                packet.position(0);
```

Comment:
Can we assume that this position will always be zero, or do we need to store the initial position before calling inputRecord.decode?

Reply:
I think we can't, good catch. It is unlikely that the alert will be following another command in the buffer but it is technically possible. I'll make a change.

Reply:
Another consideration: we won't be handling this error when `SSLSocket` is being used